pmlc/dialect/tile/ir/ops.cc

```diff
@@ -387,9 +387,27 @@ struct SymbolicContractionCanonicalizer
     auto sourceType = op.result().getType().cast<RankedTensorType>();
     auto resultType = RankedTensorType::get(shape, sourceType.getElementType());
     if (!resultType.hasStaticShape()) {
-      return failure();
+      if (resultType == sourceType) {
+        return failure();
+      }
+      // Can't rewrite to a non-symbolic contraction, but we can at least update
+      // the result type
+      auto newOp = rewriter.create<SymbolicContractionOp>(
+          op.getLoc(), resultType, op.init(), op.cons(), op.size(), op.sink(),
+          op.srcs(), op.agg(), op.combo(), rewriter.getUnitAttr(),
+          rewriter.getStringAttr(op.name().getValueOr("")));
+      // TODO: This seems like not the right way to handle no_reduce
+      if (op.no_reduce().hasValue()) {
+        newOp.setAttr("no_reduce", rewriter.getUnitAttr());
+      } else {
+        newOp.removeAttr("no_reduce");
+      }
```

Comment:
I think you can say:

```
      auto nameAttr = rewriter.getStringAttr(op.name().getValueOr(""));
      auto reduceAttr = op.no_reduce().hasValue() ? rewriter.getUnitAttr() : UnitAttr{};
      auto newOp = rewriter.create<SymbolicContractionOp>(
          op.getLoc(), resultType, op.init(), op.cons(), op.size(), op.sink(),
          op.srcs(), op.agg(), op.combo(), reduceAttr,
          nameAttr);
```
```

Reply:
Great, thanks!

I realize I don't currently have an easy way to confirm this does the same thing, so I'll be adding a lit test with these changes as well.